frontend/src/components/taskSelection/index.js

```diff
@@ -1,133 +1,114 @@
-      this.getProjectTasks();
-    }
-  }
-  componentDidUpdate(prevProps) {
-    if (this.props.project.projectId && this.props.project.projectId !== prevProps.project.projectId) {
-      this.getProjectTasks();
-    }
-  }
-  getProjectTasks = event => {
-    this.getProjectTasksPromise = cancelablePromise(
-      fetchLocalJSONAPI(`projects/${this.props.project.projectId}/tasks/`)
-    );
-    this.getProjectTasksPromise.promise
-      .then(data => this.setState({tasks: data}))
-      .catch(e => console.log(e));
-  }
-  renderHeaderLine() {
+function HeaderLine({author, projectId, priority}: Object) {
     const userLink = (
-      <Link to={`/users/${this.props.project.author}`} className="link blue-dark underline">
-        {this.props.project.author}
+      <Link to={`/users/${author}`} className="link blue-dark underline">
+        {author}
       </Link>
     );
     return (
       <div className="cf">
         <div className="w-70 dib fl">
-          <span className="blue-light">#{this.props.project.projectId}</span>
+          <span className="blue-light">#{projectId}</span>
           <span className="blue-dark">
             {' '}
             | <FormattedMessage {...messages.createBy} values={{ user: userLink }} />
           </span>
         </div>
         <div className="mw4 dib fr">
-          <PriorityBox priority={this.props.project.projectPriority} extraClasses={'pv2 ph3'} />
+          <PriorityBox priority={priority} extraClasses={'pv2 ph3'} />
         </div>
       </div>
     );
   }
 
-  render() {
-    return (
-      <div>
-        <div className="cf pv3">
-          <div className="w-100 w-60-ns fl">
-            <div className="ph4">
-              <ReactPlaceholder
-                showLoadingAnimation={true}
-                rows={3}
-                delay={500}
-                ready={typeof(this.props.project.projectId) === 'number' && this.props.project.projectId > 0}
-              >
-                {this.renderHeaderLine()}
-                <div className="cf pb3">
-                  <h3 className="f2 fw6 mt2 mb3 ttu barlow-condensed blue-dark">
-                    {this.props.project.projectInfo && this.props.project.projectInfo.name}
-                  </h3>
-                  <span className="blue-light">{this.props.project.campaignTag} &#183; Brazil</span>
+export function TaskSelection({project, type, loading}: Object) {
+  const [activeSection, setActiveSection] = useState('tasks');
+  const defaultEditor = useSelector(state => state.preferences.default_editor);
+  const [error, tasksLoading, tasks] = useFetch(`projects/${project.projectId}/tasks/`);
+
+  return (
+    <div>
+      <div className="cf pv3">
+        <div className="w-100 w-50-ns fl">
+          <div className="ph4">
+            <ReactPlaceholder
+              showLoadingAnimation={true}
+              rows={3}
+              delay={500}
+              ready={typeof(project.projectId) === 'number' && project.projectId > 0}
```

Comment:
The best way to do these `ready={}` is to use the `!loading` state that is coming in from `useFetch` – I think I tried it and it wasn't completely reliable though. Not sure why.

Reply:
On that case, the projectId is not coming from the useFetch request. I made the verification that way to solve some lint warnings